sdk/nodejs/output.ts

```diff
@@ -778,7 +778,12 @@ export type Lifted<T> =
     // Specially handle 'string' since TS doesn't map the 'String.Length' property to it.
     T extends string ? LiftedObject<String, NonFunctionPropertyNames<String>> :
     T extends Array<infer U> ? LiftedArray<U> :
-    LiftedObject<T, NonFunctionPropertyNames<T>>;
+    T extends object ? LiftedObject<T, NonFunctionPropertyNames<T>> :
+    // fallback to lifting no properties.  Note that `Lifted` is used in
+    //    Output<T> = OutputInstance<T> & Lifted<T>
+    // so returning an empty object just means that we're adding nothing to Output<T>.
+    // This is needed for cases like `Output<any>`.
+    {};
```

Comment:
this happened due to the removal of `T extends Resource ? { } : ...` that came in: https://github.com/pulumi/pulumi/commit/ad4da29cbddee7d9ddfb59f638cb3375e6c15b07#diff-b7a9c3c1793336a377df01cb5e0e9f40

Previously, the `any` cases in `Output<any>` would be caught by this, effectively adding no members to `Output<any>`.   i.e. `Output<any> = OutputInstance<any> & {}` TS seems to have refined things slightly so that this now becomes `Output<any> = OutputInstance<any> & Lifted<{}>` which it feels is no longer compatible because objects are indexable.  

My theory is that this fell out of TS 3.7's [recursive type aliases](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#more-recursive-type-aliases) work (something which we could benefit from).  Basically, instead of aggressively flattening the type, they lazily do it, but it means that they can see different types here which now triggers some subtle assignability issue.

Regardless, the above should be strictly better and works downlevel and on 3.7.  It does really indicate what we want, namely that only 'objects/arrays' should have their properties lifted up, and anything else should not participate in lifting.